src/prefect/deployments/schedules.py

```diff
@@ -1,21 +1,17 @@
-from typing import List, Optional, Sequence, Union, get_args
+from typing import TYPE_CHECKING, List, Optional, Sequence, Union, get_args
 
 from prefect.client.schemas.objects import MinimalDeploymentSchedule
-from prefect.client.schemas.schedules import SCHEDULE_TYPES
 
-try:
-    from prefect.server.schemas.schedules import SCHEDULE_TYPES as SERVER_SCHEDULE_TYPES
+if TYPE_CHECKING:
+    from prefect.client.schemas.schedules import SCHEDULE_TYPES
```

Comment:
@bunchesofdonald Are you able to comment on why this one is currently `prefect.server`? It seemed to be backwards compatibility for `build_from_flow` based on the PR I saw, but wanted to double check on it

Reply:
Yeah, essentially. Since we're doing an `isinstance` check we need to check against the server's version for that to work out, the client's version isn't the same _instance_. Having this behind a `TYPE_CHECKING` flag should be fine.

Reply:
Okay, cool, what I'm hearing is that we should then probably do:

```python
if TYPE_CHECKING:
    from prefect.server.schemas.schedules import SCHEDULE_TYPES
```

Reply:
Hello!
Currently (Before this PR), we have:
```python
from prefect.client.schemas.schedules import SCHEDULE_TYPES
```

`SERVER_SCHEDULE_TYPES` (prefect.server.schemas.schedules) is only used to raise an error.

Why would we want to replace this line with the server version?
Is the current code no good?

Reply:
@bsignoret Ah, no that's not what I meant, I meant in addition to that, but looking further, keeping it behind `TYPE_CHECKING` won't work anyway.

However, I think we could simplify this with reflecting type checking like so:
```python
is_server_schema = obj.__class__.__module__ == "prefect.server.schemas.schedules"
if is_server_schema:
    raise ValueError(
        "Server schema schedules are not supported. Please use "
        "the schedule objects from `prefect.client.schemas.schedules`"
    )
```

Then we can avoid importing any server-side schema, even behind a type check. Would you be willing to give that a shot? Tests passed for me locally with that change.

Reply:
Thank you for your idea. I've just made the modification and tests are also passed on my side.

Reply:
Nice! I'm running our workflows again, and if they pass, I think to future-proof against `schedule` module name changes and to make it more generic, we probably want to change this last line:
`return obj.__class__.__module__.startswith("prefect.server.schemas")`

Reply:
Sorry, hope you don't mind but I'd love to get this in for Thursday's release, so I made this change in [cc8dde8](https://github.com/PrefectHQ/prefect/pull/13281/commits/cc8dde89ab427f173ad2743744dbeadd5a4c375b)